src/main/java/TaskList.java

```diff
@@ -0,0 +1,132 @@
+package duke;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Represents the list of task managed by the ChatBot.
+ */
+public class TaskList {
+
+    private List<Task> list;
+    private static int noOfTasks;
+    private Ui ui;
+    private static HashMap<String, List<Task>> taskByKeyword;
+
+    /**
+     * Create a tasklist.
+     * @param pastList
+     */
+    TaskList(List<Task> pastList) {
+        this.list = pastList;
+        this.noOfTasks = pastList.size();
+        this.ui = new Ui();
+        this.taskByKeyword = new HashMap<String, List<Task>>();
+        for (int i = 0; i < pastList.size(); i++) {
+            Task pastTask = pastList.get(i);
+            String stringDescription = pastTask.getTaskDescription();
+            updateKeyword(stringDescription, pastTask);
+        }
+    }
+
+    /**
+     * Add new task from the users into the existing taskList.
+     * Keep track of the number of tasks in the list.
+     * Update the keyword hashtable.
+     * @param Task task: Task given by the users to add into the list.
+     * @return int noOfTasks: the current number of tasks in the list after adding it to the list.
+     */
```

Comment:
Perhaps the comments would be easier to read with a blank line between the method description and the @params?